src/hyp_layout.rs

```diff
@@ -75,6 +86,17 @@ const_assert!(PageSize::Size4k.is_aligned(UMODE_INPUT_START));
 /// Size of the U-mode Input Region.
 pub const UMODE_INPUT_SIZE: u64 = 4 * 1024;
 const_assert!(PageSize::Size4k.is_aligned(UMODE_INPUT_SIZE));
+/// End of the U-mode Input Region.
+const UMODE_INPUT_END: u64 = UMODE_INPUT_START + UMODE_INPUT_SIZE;
+/// Last U-mode address.
+const UMODE_END: u64 = UMODE_INPUT_END;
+
+/// The lowest possible kernel stack address.
+pub const HYP_STACKBTM_MIN: u64 = UMODE_END + 4 * 1024 * 1024;
+/// Address of the hypervisor stack top.
+pub const HYP_STACKTOP: u64 = 0xffff_ffff_ffff_f000;
```

Comment:
i'd make the stack alignment something greater than 4kb. say 16kb, or 2MB, in order to be able to take advantage of Svnapot or (unlikely) huge PTEs.

Reply:
Aligned at 2M.